cfy_manager/components/prometheus/prometheus.py

```diff
@@ -0,0 +1,357 @@
+#########
+# Copyright (c) 2020 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+
+from os import sep
+from os.path import join
+
+import argh
+
+from ..base_component import BaseComponent
+from ..components_constants import (
+    CONFIG,
+)
+from ..service_names import (
+    PROMETHEUS,
+    NODE_EXPORTER,
+    BLACKBOX_EXPORTER,
+    POSTGRES_EXPORTER,
+    RABBITMQ_EXPORTER,
+)
+from ... import constants
+# from ...config import config
+from ...constants import (
+    CLOUDIFY_USER,
+    CLOUDIFY_GROUP
+)
+# from ...exceptions import NetworkError
+from ...logger import get_logger, setup_console_logger
+from ...utils import common, files, service
+
+# from ...utils.network import wait_for_port
+# from ...utils.network import is_port_open
+
+LOG_DIR = join(constants.BASE_LOG_DIR, PROMETHEUS)
+BIN_DIR = join(sep, 'usr', 'local', 'bin')
+CONFIG_PATH = join(constants.COMPONENTS_DIR, PROMETHEUS, CONFIG)
+PROMETHEUS_CONFIG_DIR = join(sep, 'etc', 'prometheus', )
+PROMETHEUS_CONFIG_PATH = join(PROMETHEUS_CONFIG_DIR, 'prometheus.yml')
+SYSTEMD_CONFIG_DIR = join(sep, 'etc', 'systemd', 'system')
+PROMETHEUS_VERSION = '2.18.1'
+NODE_EXPORTER_VERSION = '1.0.0'
+BLACKBOX_EXPORTER_VERSION = '0.16.0'
+POSTGRES_EXPORTER_VERSION = '0.8.0'
+RABBITMQ_EXPORTER_VERSION = '1.0.0-RC7'
+PROMETHEUS_DATA_DIR = join(sep, 'var', 'lib', 'prometheus')
+
+logger = get_logger(PROMETHEUS)
+
+
+class Prometheus(BaseComponent):
+    component_name = 'prometheus'
+
+    def install(self):
+        _install_prometheus()
+        _install_node_exporter()
+        _install_blackbox_exporter()
+        _install_postgres_exporter()
+        _install_rabbitmq_exporter()
```

Comment:
@mateumann maybe these methods should be static methods ?

Reply:
I wrote them as standalone functions inside `prometheus.py`.  In my opinion Python package/module structure easily indicates those are related to Prometheus.  (And since those functions do not use `Prometheus` class objects, I did not see any good reason to have those implemented as part of `Prometheus` class.)

All in all, I guess, that's a aesthetic question :)